ui/library/test/utils/scroll.test.ts

```diff
@@ -182,3 +175,20 @@ describe('scrollToPosition', () => {
     targetDiv.remove();
   });
 });
+
+describe('calculateTop', () => {
+  it('should calculate correct value', () => {
+    expect(calculateTop(1104, 0, 24, 1056, 282.156)).equal(797);
+  });
+});
+
+describe('getScrollParent', () => {
+  it('should return a scrollable element', () => {
+    const page1 = document.createElement('div');
+    page1.id = generatePageIdFromIndex(0);
+    page1.style.height = '700px';
+    page1.style.width = '500px';
+    page1.style.margin = '20px';
+    expect(getScrollParent(page1).innerHTML).to.deep.equal(document.documentElement.innerHTML);
```

Comment:
The failure case here is really weird. It'll output a ton of HTML, which doesn't help to figure out why it's broken. A better bet would be to look at `.nodeName`.